dist/index.js

```diff
@@ -255,6 +279,39 @@ var encodeV1 = (key, value) => {
       len.writeUInt16BE(encodedInteractionTypes.byteLength);
       return Buffer.concat([tagBuf, len, encodedInteractionTypes]);
     }
+    case "marketable":
+    case "stables":
+    case "ethTransfers":
+    case "defiSwap": {
+      let encodedHomepagePending = Buffer.allocUnsafe(0);
+      Object.entries(value).forEach(([key2, value2]) => {
+        const encoded = encodeV1(key2, value2);
+        if (encoded) {
+          encodedHomepagePending = Buffer.concat([
+            encodedHomepagePending,
+            encoded
+          ]);
+        }
+      });
+      const len = Buffer.allocUnsafe(2);
+      len.writeUInt16BE(encodedHomepagePending.byteLength);
+      return Buffer.concat([tagBuf, len, encodedHomepagePending]);
+    }
+    case "optimisticL2": {
```

Comment:
is optimistcL2 same as case above (could be combined)?

Reply:
They are different types and have to be split up